statsd/uds.go

```diff
@@ -29,11 +43,74 @@ func newUdsWriter(addr string) (*udsWriter, error) {
 	if err != nil {
 		return nil, err
 	}
-	// Defer connection to first Write
-	writer := &udsWriter{addr: udsAddr, conn: nil, writeTimeout: defaultUDSTimeout}
+
+	writer := &udsWriter{
+		addr:                       udsAddr,
+		conn:                       nil,
+		writeTimeout:               defaultUDSTimeout,
+		datagramQueue:              make(chan []byte, 8192),
+		stopChan:                   make(chan struct{}, 128),
+		datagramsDroppedOutputChan: make(chan struct{}, 128),
+		datagramsDroppedQueueChan:  make(chan struct{}, 128),
+		datagramsTotalChan:         make(chan struct{}, 128),
```

Comment:
Curious why we are using a chan here rather than a mutex or atomic values as it's a buffered chan this would be blocking if the telemetry loop is stuck?

Reply:
Honestly I just thought it was the most go-like solution. However after some benchmarking I was actually planning to replace those by Atomic values which will probably have less impact.